projects/pastanaga-angular/src/lib/controls/textfield/input/input-component-tests/input-component-form-control-name.spec.ts

```diff
@@ -0,0 +1,248 @@
+            [min]="min"
+            [max]="max"
+            [maxlength]="maxlength"
+            [noAutoComplete]="noAutoComplete"
+            [acceptHtmlTags]="acceptHtmlTags"
+            [debounceDuration]="debounceDuration"
+            (valueChange)="onValueChange($event)"
+            (debouncedValueChange)="onDebouncedValueChange($event)"
+            (keyUp)="onKeyup($event)"
+            (enter)="onEnter($event)"
+            (focusing)="onFocusing($event)"
+            (blurring)="onBlurring($event)"
+            (ngModelChange)="onNgModelChange($event)"
+            >Label
+        </pa-input>
+    </form>`,
+})
+export class TestComponent {
+    @ViewChild('paInput') paField?: InputComponent;
+    form = new FormGroup({
+        text: new FormControl(),
+    });
+    get formControl() {
+        return this.form.get('text');
+    }
+    id?: string;
+    name?: string;
+    help?: string;
+    describedBy?: string;
+    readonly = false;
+    hasFocus = false;
+    showAllErrors = false;
+    errorMessages?: any;
+    errorMessage?: string;
+    updateOn: UpdateOnStrategy = 'change';
+    value?: string | number;
+    type?: TextInputType;
+    placeholder?: string;
+    required = false;
+    pattern?: string | RegExp;
+    min?: number;
+    max?: number;
+    maxlength?: number;
+    noAutoComplete = false;
+    acceptHtmlTags = false;
+    // avoid debouncing for most input-component-tests
+    debounceDuration? = 0;
+
+    onValueChange(event: any) {}
+
+    onDebouncedValueChange(event: any) {}
+
+    onKeyup(event: any) {}
+
+    onEnter(event: { event: KeyboardEvent; value: any }) {}
+
+    onFocusing(event: FocusEvent) {}
+
+    onBlurring(event: any) {}
+
+    onNgModelChange(event: any) {}
+}
+
+describe('InputFieldComponent FormControlName', () => {
+    let nextId = 0;
+    let fixture: ComponentFixture<TestComponent>;
+    beforeEach(async(() => {
+        nextId++;
+        fixture = initTest(TestComponent);
+    }));
+
+    it('should assign id', fakeAsync(() => testId(fixture, nextId)));
+
+    it('should assign name', fakeAsync(() => testName(fixture, nextId)));
+
+    it('should assign help', fakeAsync(() => testHelp(fixture)));
+
+    it('should assign describedBy', fakeAsync(() => testDescribedBy(fixture, nextId)));
+
+    it('should assign readonly', fakeAsync(() => testReadonly(fixture)));
+
+    it('should focus field', fakeAsync(() => testFocus(fixture)));
+
+    it('should apply disabled state', fakeAsync(() => {
+        clearFakeAsyncZone(fixture);
+
+        fixture.componentInstance.form.disable();
+        fixture.detectChanges();
+        tick();
+        thenFieldControlIsDisabled(fixture, true);
+
+        fixture.componentInstance.form.enable();
+        fixture.detectChanges();
+        tick();
+        thenFieldControlIsDisabled(fixture, false);
+
+        if (!fixture.componentInstance.formControl) {
+            expect(true).toEqual(false);
+            return;
+        }
```

Comment:
why not just:
```
expect(fixture.componentInstance.formControl).toBeTruthy();
```

Reply:
c'est just un hack pour faire un null check avec erreur... si y a pas de formControl, c'est pas sensé arriver et ça sert à rien de passer à la suite...

Reply:
sorry but I don't get it. You just make the test failing when there is no `fixture.componentInstance.formControl` with that code. so from what I understand doing `expect(fixture.componentInstance.formControl).toBeTruthy();` would have exactly the same result as it would fail the test if there is no `fixture.componentInstance.formControl` and the following expectations would not be executed.